stakingModel/contracts/ContributionCalculation.sol

```diff
@@ -0,0 +1,67 @@
+pragma solidity 0.5.4;
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "@daostack/arc/contracts/controller/Avatar.sol";
+import "../../contracts/dao/schemes/SchemeGuard.sol";
+import "../../contracts/DSMath.sol";
+import "../../contracts/token/GoodDollar.sol";
+
+/* @title Contribution calculation for selling gd tokens
+ */
+contract ContributionCalculation is DSMath, SchemeGuard {
+    using SafeMath for uint256;
+
+    GoodDollar gooddollar;
+    uint256 public sellContributionRatio;
+
+    event SellContributionRatioUpdated(address indexed caller,
+                                       uint256 nom,
+                                       uint256 denom);
+
+    constructor(
+        Avatar _avatar,
+        address _gooddollar,
+        uint256 _nom,
+        uint256 _denom
+    )
+        public
+        SchemeGuard(_avatar)
+    {
+        gooddollar = GoodDollar(_gooddollar);
+        sellContributionRatio = rdiv(_nom, _denom);
+    }
+
+    /**
+    @dev allow the DAO to change the sell contribution rate
+    it is calculated by _nom/_denom with e27 precision
+    @param _nom the nominator
+    @param _denom the denominator
+    */
+    function setContributionRatio(uint256 _nom, uint256 _denom)
+        external
+        onlyAvatar
+    {
+        sellContributionRatio = rdiv(_nom, _denom);
+        emit SellContributionRatioUpdated(msg.sender, _nom, _denom);
+    }
+
+    /**
+    * @dev calculate the contribution amount during the sell action. there is a
+    * `sellContributionRatio` percent contribution
+    * @return (contributionAmount) the contribution amount for sell
+    */
+    function calculateContribution(uint256 gdAmount)
```

Comment:
i think token being converted should also be passed (currently its only cdai)

Reply:
added ea7fa3b9d47af8ec3aead553cd095899c8c4da66